spark-connect/index.md

```diff
@@ -0,0 +1,151 @@
+
+**Executing the query on the Spark Server**
+
+The Spark Server receives the unresolved logical plan (once the Protocol Buffer is deserialized) and executes it just like any other query.
+
+Spark performs many optimizations to an unresolved logical plan before executing the query.  All of these optimizations happen on the Spark Server.
+
+Spark Connect lets you leverage Spark’s powerful query optimization capabilities, even with Clients that don’t depend on Spark or the JVM.
+
+**Sending the results back to the Client**
+
+The Spark Server sends the results back to the Client after executing the query.
+
+The results are sent to the client as Apache Arrow record batches.  A record batch includes many rows of data.
+
+The record batch is streamed to the client, which means it is sent in partial chunks, not all at once.  Streaming the results from the Spark Server to the Client prevents memory issues caused by an excessively large request.
+
+Here’s a recap of how Spark Connect works in image form:
+
+<img src="{{site.baseurl}}/images/spark-connect2.png" style="width: 100%; max-width: 500px;">
+
+## Benefits of Spark Connect
+
+Let’s now turn our attention to the benefits of the Spark Connect architecture.
+
+**Spark Connect workloads are easier to maintain**
+
+With the Spark JVM architecture, the client and Spark Driver must run identical software versions.  They need the same Java, Scala, and other dependency versions.  Suppose you develop a Spark project on your local machine, package it as a JAR file, and deploy it in the cloud to run on a production dataset.  You need to build the JAR file on your local machine with the same dependencies used in the cloud.  If you compile the JAR file with Scala 2.13, you must also provision the cluster with a Spark JAR compiled with Scala 2.13.
+
+Suppose you are building your JAR with Scala 2.12, and your cloud provider releases a new runtime built with Scala 2.13.  For Spark JVM, you need to update your project locally, which may be challenging.  For example, when you update your project to Scala 2.13, you must also upgrade all the project dependencies (and transitive dependencies) to Scala 2.13.  If some of those JAR files don’t exist, you can’t upgrade.
+
+In contrast, Spark Connect decouples the client and the Spark Driver, so you can update the Spark Driver including server-side dependencies without updating the client.  This makes Spark projects much easier to maintain.
+
+**Spark Connect lets you build Spark Connect Clients in non-JVM languages**
+
+Spark Connect decouples the client and the Spark Driver so that you can write a Spark Connect Client in any language.  Here are some Spark Connect Clients that don’t depend on Java/Scala:
+
+* [Spark Connect Python](https://github.com/apache/spark/tree/master/python/pyspark/sql/connect)
+* [Spark Connect Go](https://github.com/apache/spark-connect-go)
+* [Spark Connect Rust](https://github.com/sjrusso8/spark-connect-rs)
```

Comment:
To @srowen , is it okay we add this link here in Apache Spark website officially?

It seems to make the readers confused this as one of the Apache Spark community stuff.

Shall we remove this link because the Python and Go is enough to show the rational of this paragraph?